### PR TITLE
feat(molgenis): add oidc sidecar

### PR DESCRIPTION
**Pull request by sidohaakma**
#### Checklist
- [x] Functionality works & meets specifications (tested on rancher)
- [x] Templates reviewed
- [x] Added values to questions
- [x] Bumped the chart version
- [x] Updated appVersion (if needed)

### CODE REVIEW

**Comment by sidohaakma** [comment]
[Add OIDC config to helm chart](https://trello.com/c/W1f2hYPv/2-add-oidc-config-to-helm-chart)

**Comment by tommydeboer** [comment]
add some documentation in the question

File: charts/molgenis/questions.yml [hunk 1]
```diff
@@ -35,6 +35,92 @@ questions:
...
+    group: "Provisioning - OIDC"
+  - variable: oidc.identification.endpoints.userinfo
+    label: User information endpoint
+    default: "https://auth.molgenis.org/oauth2/userinfo"
+    description: "Specify the user information endpoint of the ID-provider"
+    type: string
+    required: true
+    show_if: "oidc.enabled=true"
+    group: "Provisioning - OIDC"
+  - variable: oidc.identification.endpoints.userinfo
```
Comment (fdlk) [inline charts/molgenis/questions.yml:93]: copy paste error, should be jwks

File: charts/molgenis/questions.yml [hunk 2]
```diff
@@ -35,6 +35,92 @@ questions:
...
+    group: "Provisioning - OIDC"  
+  - variable: oidc.identification.client_secret
+    label: Client secret
+    default: ""
+    description: "Type in the client secret"
+    type: password
+    required: true
+    show_if: "oidc.enabled=true"
+    group: "Provisioning - OIDC"
+  - variable: oidc.identification.endpoints.authorisation
```
Comment (fdlk) [inline charts/molgenis/questions.yml:77]: Starting 8.7, the endpoints can be discovered if you provide the issuer URL, propose to only ask for that in the questions

File: charts/molgenis/templates/oidc-secret.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{{ if .Values.oidc.enabled -}}
```
Comment (fdlk) [inline charts/molgenis/templates/oidc-secret.yaml:1]: Nice to make it a secret!

File: charts/molgenis/values.yaml [hunk 4]
```diff
@@ -46,6 +46,31 @@ resources:
 catalinaOpts: "-XX:MaxRAMPercentage=80 -XX:+UseConcMarkSweepGC -XX:+CMSClassUnloadingEnabled"
 maxFileMB: 150
 
+oidc:
+  enabled: true
+  id: molgenis-auth
+  name: molgenis-auth
+  identification:
+    endpoints:
+      authorization: https://auth.molgenis.org/oauth2/authorize
```
Comment (fdlk) [inline charts/molgenis/values.yaml:55]: I'd say specify only issuerURI, unless we really want backwards compatibility with 8.6

File: charts/molgenis/Chart.yaml [hunk 5]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: "8.7"
 description: MOLGENIS - helm stack
 name: molgenis
-version: 1.13.1
+version: 1.14.0
```
Comment (fdlk) [inline charts/molgenis/Chart.yaml:5]: 1.15.0 now, @tommydeboer and I skipped the queue

File: charts/molgenis/questions.yml [hunk 6]
```diff
@@ -35,6 +35,92 @@ questions:
     type: string
     required: true
     group: "Provisioning"
+  - variable: oidc.enabled
+    label: OIDC enabled
+    default: true
+    description: "Do you want to connect to and ID-provider or authentication service"
```
Comment (fdlk) [inline charts/molgenis/questions.yml:41]: and -> an

File: charts/molgenis/questions.yml [hunk 7]
```diff
@@ -35,6 +35,92 @@ questions:
...
+    type: boolean
+    required: true
+    group: "Provisioning - OIDC"
+  - variable: oidc.id
+    label: OpenID connect ID
+    default: "molgenis-auth"
+    description: "Type openid connect ID"
```
Comment (fdlk) [inline charts/molgenis/questions.yml:48]: Explain that it ends up in the callback URL, see OidcClient metadata

File: charts/molgenis/questions.yml [hunk 8]
```diff
@@ -35,6 +35,92 @@ questions:
...
+    type: string
+    required: true
+    show_if: "oidc.enabled=true"
+    group: "Provisioning - OIDC"
+  - variable: oidc.name
+    label: OpenID connect name
+    default: "molgenis-auth"
+    description: "Type openid connect name"
```
Comment (fdlk) [inline charts/molgenis/questions.yml:56]: Explain that it ends up as label of the login button, see OidcClient metadata

File: charts/molgenis/questions.yml [hunk 9]
```diff
@@ -35,6 +35,92 @@ questions:
...
+    type: string
+    required: true
+    show_if: "oidc.enabled=true"
+    group: "Provisioning - OIDC"  
+  - variable: oidc.identification.client_id
+    label: Client ID
+    default: ""
+    description: "Type in the client id"
```
Comment (fdlk) [inline charts/molgenis/questions.yml:64]: Explain that it is provided by the id provider

File: charts/molgenis/questions.yml [hunk 10]
```diff
@@ -35,6 +35,92 @@ questions:
...
+    type: string
+    required: true
+    show_if: "oidc.enabled=true"
+    group: "Provisioning - OIDC"  
+  - variable: oidc.identification.client_secret
+    label: Client secret
+    default: ""
+    description: "Type in the client secret"
```
Comment (fdlk) [inline charts/molgenis/questions.yml:72]: Explain that it is provided by the ID provider

File: charts/molgenis/questions.yml [hunk 11]
```diff
@@ -35,6 +35,92 @@ questions:
...
+    label: Commander enabled
+    default: true
+    description: "Enable the commander to bootstrap your MOLGENIS"
+    type: boolean
+    required: true
+    group: "Provisioning - Commander"
+  - variable: commander.script
+    label: Commander script
+    default: ""
+    description: "Give the commander script as it was a script"
```
Comment (fdlk) [inline charts/molgenis/questions.yml:119]: Probably can be formulated a bit more crisply

File: charts/molgenis/values.yaml [hunk 12]
```diff
@@ -46,6 +46,31 @@ resources:
...
+  claims:
+    roles: roles
+    vo_group:
+
+commander:
+  enabled: false
+  image:
+    repository: molgenis
+    name: commander
+    tag: latest
```
Comment (fdlk) [inline charts/molgenis/values.yaml:70]: Provide a question for this one I think

File: charts/molgenis/templates/commander-job.yaml [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+          image: fabric8/fabric8-dependency-wait-service
+          env:
+            - name: MOLGENIS_HOST
+              value: {{ template "molgenis.fullname" . }}
+            - name: DEPENDENCY_POLL_INTERVAL
+              value: "11"
+            - name: DEPENDENCY_LOG_VERBOSE
+              value: "true"
+          command: ["/bin/sh"]
+          args: ["-c", "fabric8-dependency-wait-service-linux-amd64 http://$(MOLGENIS_HOST):8080"]
```
Comment (fdlk) [inline charts/molgenis/templates/commander-job.yaml:22]: this one comes up quite early, while bootstrapping is still in place, I think. In the startup prompt we use the `/api/v2/version` endpoint instead

**Review by fdlk — CHANGES_REQUESTED** [review]
Very much looking forward to this feature!

File: charts/molgenis/values.yaml (hunk 12 above)
Comment (sidohaakma) [reply charts/molgenis/values.yaml:70]: For the version you mean? Or both? The repo and the version?

File: charts/molgenis/questions.yml (hunk 11 above)
Comment (sidohaakma) [reply charts/molgenis/questions.yml:119]: Hahaha, yes...

File: charts/molgenis/values.yaml (hunk 12 above)
Comment (fdlk) [reply charts/molgenis/values.yaml:70]: version I think

File: charts/molgenis/templates/commander-job.yaml (hunk 13 above)
Comment (fdlk) [reply charts/molgenis/templates/commander-job.yaml:22]: Yes, so you wait until bootstrapping is over

File: charts/molgenis/questions.yml (hunk 1 above)
Comment (sidohaakma) [reply charts/molgenis/questions.yml:93]: Removed in favor of issuerUri

File: charts/molgenis/questions.yml (hunk 2 above)
Comment (sidohaakma) [reply charts/molgenis/questions.yml:77]: Removed in favor of issuerUri

**Review by fdlk — APPROVED** [review]

**Pull request merged**